### ISSUE TITLE
Define a way to pre-select a path and file name

### ISSUE THREAD

**Issue by aideenf** [crahan/ipyfilechooser#27]
Hi there
I would like to have the ability to predefine a selected path and a selected file name.
Now I  use:

fc = FileChooser()
fc.default_path = os.getcwd() #current directory (using import 0s)
fc.default_filename = 'transformed_file.csv' 
display (fc.default)
'/Users/aideenf/Documents/GitHub/Thesis_Aideen/Main/transformed_19_08_2020.csv'

which is perfect, however, I would like to set

fc.selected = fc.default

so that when I render the widget the specific path and file is already selected and the control is in
"change" mode rather than in "select" mode.

Regards
Aideen

**Comment by aideenf** [comment]
Actually I just looked through the code and I see that fc._apply_selection() achieves exactly what I was looking for.

Thank you!

**Comment by crahan** [comment]
@aideenf, I think the quickest option is probably to use the `select_default` parameter:
```
fdialog = FileChooser(
    os.getcwd(),
    filename='transformed_file.csv',
    select_default=True
)
```

**Comment by aideenf** [comment]
@crahan That does look a lot neater! Thanks a lot for the quick reply.

Regards,
Aideen

**Comment by CptofMySoul** [comment]
The `select_default=True` parameter only seems to work when the `FileChooser` is first created.

I'm doing this:
```
fc.reset(filename="new_file")
fc._apply_selection()
```
in order to change the selected file programmatically.  Is there a better way?

**Comment by crahan** [comment]
@CptofMySoul would the following be a solution to your use case?

```
fdialog.default_filename='hello.md'
fdialog.reset()
```

**Comment by crahan** [comment]
@CptofMySoul that would be the best way to do this right now. select_default was implemented mainly as a way to have a default value selected upon creation of a `FileChooser` object.

It sounds like there might be a need to programmatically change and apply the selected path and filename. Can you submit a new issue for this so I can properly track it for the next update?

**Comment by crahan** [comment]
@CptofMySoul I've merged the fix (https://github.com/crahan/ipyfilechooser/pull/33) and released version 0.4.1 which fixes this. Thanks again for pointing out the bug! Much appreciated.
